server/src/main/java/org/opensearch/index/SegmentReplicationRejectionStats.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index;
+
+import org.opensearch.Version;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * Segment Replication Rejection Stats.
+ *
+ * @opensearch.internal
+ */
+public class SegmentReplicationRejectionStats implements Writeable, ToXContentFragment {
```

Comment:
May be we can remove this class as we are only tracking a long variable here unless we seek future changes to add more fields.

Reply:
Yes, @mch2 suggested [here](https://github.com/opensearch-project/OpenSearch/pull/10656#discussion_r1361614250) for a new class

Reply:
Would rather we keep this so we aren't directly writing the xcontent in NodeStats